guides/plugins/plugins/content/cms/add-data-to-cms-elements.md

```diff
@@ -161,3 +161,68 @@ This could be a possible solution for that:
 ```
 
 :::
+
+### Event-based extensibility
+
+In Shopware’s CMS flow, CMS Elements are not “live bound” to the original entity (e.g. a product).
+Instead, during the slot-resolution, resolvers copy values from the entity into internal CMS structs
+(for example, `ProductNameCmsElementResolver` takes the string `name` from the `product` entity and writes it into the CMS text element).
+Once that copy is done, the storefront rendering reads from the CMS structs — not from the original entity.
+Therefore: if you wait until a “page loaded” event (e.g. `ProductPageLoadedEvent`) after the copying happened, changing the underlying entity has no effect on what is displayed in the CMS output.
+To make modifications effective (e.g. change product name, adjust a field, override some data), you must intervene before or after the resolver runs — i.e. at a point in the CMS resolution pipeline where the entity is still used for populating the CMS slots.
+
+#### Available Extensions / Events
+
+Shopware exposes three CMS extension classes under `Shopware\Core\Content\Cms\Extension`.
+These extension classes follow the common Extension-Point pattern in Shopware and publish named hooks that you can subscribe to (the classes usually expose a `NAME` constant used as the event identifier).
+All three extension points are dispatched with lifecycle suffixes such as `.pre` and `.post`, so you will typically see event names like `cms-slots-data.resolve.pre` or `cms-slots-data.resolve.post`.
+Using the `.pre` hook lets you intervene before the respective phase runs; `.post` runs after the phase finished.
+
+##### CmsSlotsDataCollectExtension
```

Comment:
There is no fifth level in Headlines for markdown in most cases. Is there a reason you already start with level 3?

Reply:
In theory, you're right. But I encountered lots of cases where they don't so, I'd just be careful with that :) 

Nice solution!